tests/unit/components/lg-dob-search-selector-test.js

```diff
@@ -0,0 +1,91 @@
+import Ember from 'ember';
+import { dobSearchOptions } from 'lion-guardians/utils/units';
+
+import {
+  moduleForComponent,
+  test
+} from 'ember-qunit';
+
+moduleForComponent('lg-dob-search-selector', 'LgDobSearchSelectorComponent', {
+  setup: function() {
+    this.container.register('view:select', Ember.Select);
+  }
+});
+
+test('it renders', function() {
+  expect(2);
+
+  // creates the component instance
+  var component = this.subject();
+  equal(component._state, 'preRender');
+
+  // appends the component to the page
+  this.append();
+  equal(component._state, 'inDOM');
+});
+
+test('it does not set start or end when no age selected', function(){
+  expect(2);
+  var component = this.subject();
+
+  component.set('selectedDobOption', null);
+  equal(component.get('selectedSearchDateEnd'), null);
+  equal(component.get('selectedSearchDateStart'), null);
+});
+
+test('it sets selected start and end for age for very young lions', function(){
+  expect(3);
+  var component = this.subject();
+  var selectedOption = dobSearchOptions[1].value;
+
+  var today = new Date();
+
+  var startDate = new Date();
+  startDate.setFullYear(today.getFullYear() - selectedOption.maximumAge);
+
+  var endDate = new Date(); //
+
+  component.set('selectedDobOption', selectedOption);
+  equal(component.get('selectedDobOption'), selectedOption);
+  equal(component.get('selectedSearchDateStart'), startDate);
+  equal(component.get('selectedSearchDateEnd'), endDate);
+});
+
+test('it sets params for age for very old lions', function(){
+  expect(3);
+  var component = this.subject();
+  var numOptions = dobSearchOptions.length;
+  var selectedOption = dobSearchOptions[numOptions - 1].value;
+
+  var today = new Date();
+
+  var startDate = new Date();
+  startDate.setFullYear(today.getFullYear() - 100);
+
+  var endDate = new Date(); //
+  endDate.setFullYear(today.getFullYear() - selectedOption.minimumAge);
+
+  component.set('selectedDobOption', selectedOption);
+  equal(component.get('selectedDobOption'), selectedOption);
```

Comment:
you can make things a little easier on yourself by adding a message (3rd argument), like so:
`equal(component.get('selectedDobOption'), selectedOption, 'the selected dob option should be equal');`

Then it is a little easier to see in the test output which specific assertion is the one failing. You don't get the generic "failed" message or whatever it is.

Furthermore, the equals helper won't give a very good diff of the issue when it is comparing two dates. If you did `equal(1,2)`, QUnit would print out something useful like "expected 1, result 2". But it doesn't do that for dates.

Finally, using hand-constructed dates here and `moment()`-constructed dates inside the component is going to make these tests subject to timing issues. I just ran them and the `selectedSearchDateStart` and `startDate` differed by 1 millisecond, causing the test to fail.

When I test dates I usually test some more easily usable artifact of the date, because date equality is hard to do right (you have milliseconds and time zones working against you). I typically use someting like `date.toTime()` so I can compare integers, or just compare the date's year and month and day individually, maybe.